spec/requests/api/blueprints_spec.rb

```diff
@@ -0,0 +1,166 @@
+RSpec.describe "Blueprints API" do
+  describe "GET /api/blueprints" do
+    it "lists all the blueprints with an appropriate role" do
+      blueprint = FactoryGirl.create(:blueprint)
+      api_basic_authorize collection_action_identifier(:blueprints, :read, :get)
+
+      run_get(blueprints_url)
+
+      expected = {
+        "count"     => 1,
+        "subcount"  => 1,
+        "name"      => "blueprints",
+        "resources" => [
+          hash_including("href" => a_string_matching(blueprints_url(blueprint.id)))
+        ]
+      }
```

Comment:
Ok, was looking at your comment from https://github.com/ManageIQ/manageiq/pull/9731, didn't know if you wanted to do something like:

```
expect(response.parsed_body).to include(expected.merge(query_result_structure(:blueprints, 1, 1))
```